mirgecom/euler.py

```diff
@@ -373,3 +380,27 @@ def get_inviscid_timestep(discr, eos, cfl, q):
 #    wavespeeds = _get_wavespeed(w,eos=eos)
 #    max_v = clmath.max(wavespeeds)
 #    return c*dt_ngf*dt_gf/max_v
+
+
+def get_extra_init(cfl, constant_cfl, initname, eosname):
+    """Get Euler-specific init messages."""
+    return (
+        f"CFL:             {cfl}\n"
+        f"Constant CFL:    {constant_cfl}\n"
+        f"Initialization:  {initname}\n"
+        f"EOS:             {eosname}\n")
```

Comment:
Related to the previous comment about `_extra_` vs. `_custom_` or similar...

These aren't exactly `Euler`-specific, but fluid-specific. Just so happens that `Euler` is our only flow module atm.  Maybe `get_flow_init`?

We should think of the quantities that would be good to have when a user asks for support to fix their simulation.  The init message should (by default) provide at a glance the essential parameters in the flow setup.  (Might be nice if it would *validate* it too, but that's another can of worms).  These are what I came up with - but there will be a lot more with viscous flows of reactive mixtures.

Some hook to allow user customization of the init message (i.e. to *add* things to it) would be good.  The whole init msg handling is pretty clunk right now, and made a bit better by your changes... but let's go all the way?

Reply:
Yeah, I guess what I was thinking is that we would have a pair of these routines for each set of governing equations (Euler, NS, heat) and let the user mix and match based on what they're running. Having even more customization than that would be cool, but I don't really have a clear vision for how to set that up.

Reply:
Sure. I agree with the intent. Euler and NS will mostly overlap however, meaning that every time you use NS one - you need to stack the Euler one and the NS one if you don't intend to repeat code.  That leads me to suggest we have a unified "flow" init msg that just spews the minimal set of flow configuration.

Re: customization - I think we could handle it exactly like the viz customization.  If the user has junk to add, just create one's own `make_status_message`.

Reply:
Another quick thought about customization: maybe the init/status msg handling routine should just take an argument that is the user's customized init/status msg function.  Then the msg handling routine could just call that an expect a return string that is the user's customized msg addition.